hopsanc/src/hopsanc.cpp

```diff
@@ -0,0 +1,275 @@
+#include "hopsanc.h"
+#include <iostream>
+
+#include "HopsanCore.h"
+#include "HopsanEssentials.h"
+#include "ComponentSystem.h"
+
+static hopsan::ComponentSystem *spCoreComponentSystem = nullptr;
+static hopsan::HopsanEssentials gHopsanCore;
+
+static double startTime, stopTime;
+
+//! @brief Prints all waiting messages
+//! @param[in] printDebug Should debug messages also be printed
+void printWaitingMessages(hopsan::HopsanEssentials& hopsanCore, bool printDebug, bool silent)
+{
+    if(silent) return;
+
+    hopsan::HString msg, type, tag;
+    while (hopsanCore.checkMessage() > 0) {
+        hopsanCore.getMessage(msg,type,tag);
+        if (type == "debug") {
+            if (printDebug) {
+                std::cout << msg.c_str() << "\n";
+            }
+        }
+        else {
+            std::cout << msg.c_str() << "\n";
+        }
+    }
+}
+
+
+int loadModel(const char* path) {
+    spCoreComponentSystem = gHopsanCore.loadHMFModelFile(path, startTime, stopTime);
+    if(!spCoreComponentSystem) {
+        std::cout << "Failed to instantiate model!\n";
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    }
+    const hopsan::HString modelName = spCoreComponentSystem->getName();
+    spCoreComponentSystem->addSearchPath(modelName+"-resources");
+    std::cout << "Loaded model: " << modelName.c_str() << "\n";
+    printWaitingMessages(gHopsanCore, false, false);
+    return 0;
+}
+
+
+//! @brief Returns specified data vector as C array
+//! @param [in] variable Variable name ("component.port.variable")
+//! @param [out] size Size of returned data array
+//! @returns Pointer to data array
+double *getDataVector(const char* variable, int &size)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded.\n";
+        size = -1;
+        return nullptr;
+    }
+
+    //Parse variable string
+    hopsan::HString varStr(variable);
+    hopsan::HVector<hopsan::HString> splitSys = varStr.split('|');
```

Comment:
We need to normalize the "syntax" for getting components,  sometimes its $ and # sometimes its | and .
Maybe not in this PR, but it would be nice to have it correct from the start.

I think that only using . as separator should be possible, just assume that each name is a component name, until its not, then check for port. Ofcourse then there is the problem if one component has the same name as a port